### PR TITLE
Add FormatMod Support for OMs

### PR DESCRIPTION
**Pull request by CamiloTD**
This is my first pull, if im doing something wrong please tell me, what i did is simple, i wanted to make PS to recognize OMs and adjust pokemon stats depending of the format, for testing i created the "Tier Shift" support.

How does it works? 

When update() is called on client-teambuilder.js i call a function named checkFormatMods(), in that function i check for existing mods and i apply it if didnt applied before. If there is no mod for the current format, then delete the current mod.

In search.js functions, instead of using global BattlePokedex, im creating a scope variable that overrides the global one, and i give it the value of the current mod or the global object depending of the situation.

This system is extensible and i'll make it more sophisticated, but... tomorrow.

No more changes.

### CODE REVIEW

File: js/client-teambuilder.js [hunk 1]
```diff
@@ -3035,26 +3046,57 @@
...
+				} else if (SRweak >= 2 || hasMove['bellydrum']) {
+					hpDivisibility = 2;
+				} else if (SRweak >= 1 || hasMove['substitute'] || hasMove['transform']) {
+					hpDivisibility = 4;
+				} else {
+					hpDivisibility = 8;
+				}
+
+				if (hpDivisibility) {
+					while (hp < 252 && evTotal < 508 && !(stat % hpDivisibility) !== hpShouldBeDivisible) {
```
Comment (Zarel) [inline js/client-teambuilder.js]: This should not be a part of your diff. Are you unfamiliar with Git? This part needs to be rebased - I'm sure someone can tell you how to if you have trouble.

File: js/client-formatmods.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+(function(exports, $){
```
Comment (Zarel) [inline js/client-formatmods.js]: This should be `(function (exports, $){`

Which is the first of around 30 things Travis CI is currently telling you is wrong. Do you know how to use `npm test`? That will give you the same errors Travis CI is giving, without needing to manually push and check this page.

File: js/client-formatmods.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		stats.def = Math.clamp(stats.def + quant, 1, 255);
+		stats.hp  = Math.clamp(stats.hp  + quant, 1, 255);
+		stats.spa = Math.clamp(stats.spa + quant, 1, 255);
+		stats.spd = Math.clamp(stats.spd + quant, 1, 255);
+		stats.spe = Math.clamp(stats.spe + quant, 1, 255);
+	}
+
+	Math.clamp = function clamp(num, min, max){
+		return num < min? min : num > max? max : num;
+	}
```
Comment (Zarel) [inline js/client-formatmods.js]: You should probably not attach new functions to native objects. That's why we dump this in Tools.

**Comment by Zarel** [comment]
In the end, though, my biggest feedback is that you should probably just use the build script for this.

**Comment by Zarel** [comment]
Specifically, that you should edit the already-existing `./githooks/build-indexes` script to include OM mods, rather than this messier system.

File: js/client-teambuilder.js (hunk 1 above)
Comment (AustinXII) [inline js/client-teambuilder.js]: Adding a check for magic guard here would be nice

File: js/client-formatmods.js (hunk 3 above)
Comment (CamiloTD) [inline js/client-formatmods.js]: Jaja okey, gonna look for some global Tool Object

File: js/client-teambuilder.js (hunk 1 above)
Comment (CamiloTD) [inline js/client-teambuilder.js]: Yeah... i never used Github, this is my first commit, so i use Sublime Text and then paste the code, idk what could be happening

Comment (Zarel) [inline js/client-teambuilder.js]: Yeah, you can't just paste the code from a different version, you have to use the same version.

Comment (Zarel) [inline js/client-teambuilder.js]: Search for a guide on rebasing or ask us for help.

**Comment by Zarel** [comment]
This has been unmaintained for around half a year and does not implement things the way we need them; closing.

**Pull request closed without merging**